### PR TITLE
[SHELL32] Implement VK_DELETE on SHBrowseForFolder

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Enable Del/Delete key to delete the selected folder in "Browse for Folder" dialog.

JIRA issue: [CORE-7592](https://jira.reactos.org/browse/CORE-7592)

You can use https://jira.reactos.org/secure/attachment/40118/getfolder.zip as a test program.

after

### CODE REVIEW

**Comment by Mouvedia** [comment]
Can you also tackle CORE-14332 while are at it?

**Comment by katahiromz** [comment]
NOTE: When the application is using our `SHBrowseForFolder`, the desktop items won't be renamed wrongly. https://jira.reactos.org/browse/CORE-16961

**Comment by yagoulas** [comment]
Let's fix things one by one. This is ready to go.

File: dll/win32/shell32/wine/brsfolder.c [hunk 1]
```diff
@@ -625,6 +625,30 @@ static HRESULT BrsFolder_Rename(browse_info *info, HTREEITEM rename)
...
+{
+    TV_ITEMW item;
+    TV_ITEMDATA *item_data;
+    SHFILEOPSTRUCTW fileop = { info->hwndTreeView };
+    WCHAR szzFrom[MAX_PATH + 1];
+
+    item.mask  = TVIF_PARAM;
+    item.mask  = TVIF_HANDLE | TVIF_PARAM;
+    item.hItem = selected_item;
+    SendMessageW(info->hwndTreeView, TVM_GETITEMW, 0, (LPARAM)&item);
```
Comment (yagoulas) [inline dll/win32/shell32/wine/brsfolder.c:640]: check return value for error

File: dll/win32/shell32/wine/brsfolder.c [hunk 2]
```diff
@@ -625,6 +625,30 @@ static HRESULT BrsFolder_Rename(browse_info *info, HTREEITEM rename)
...
+    item_data = (TV_ITEMDATA *)item.lParam;
+
+    fileop.wFunc = FO_DELETE;
+    SHGetPathFromIDListW(item_data->lpifq, szzFrom);
```
Comment (yagoulas) [inline dll/win32/shell32/wine/brsfolder.c:644]: check return value for error

File: dll/win32/shell32/wine/brsfolder.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/wine/brsfolder.c]: Added.

File: dll/win32/shell32/wine/brsfolder.c (hunk 2 above)
Comment (katahiromz) [reply dll/win32/shell32/wine/brsfolder.c]: Added.

**Comment by yagoulas** [comment]
Good, you can merge it if it works.

**Comment by Mouvedia** [comment]
@yagoulas CORE-14332 needs to be updated. It seems like @katahiromz fixed half of it.

**Pull request merged**